src/app/my-rewards/tx-history/components/Table/Cells.tsx

```diff
@@ -0,0 +1,216 @@
+import { FC, HtmlHTMLAttributes, ReactElement, ReactNode } from 'react'
+import { cn, truncate } from '@/lib/utils'
+import { useTableContext } from '@/shared/context'
+import {
+  ColumnId,
+  COLUMN_TRANSFORMS,
+  GroupedTransactionDetail,
+  TransactionHistoryCellDataMap,
+} from './TransactionHistoryTable.config'
+import { Paragraph } from '@/components/Typography'
+import { Jdenticon } from '@/components/Header/Jdenticon'
+import { ArrowUpIcon } from '@/components/Icons/ArrowUpIcon'
+import { ArrowDownIcon } from '@/components/Icons/ArrowDownIcon'
+import { TokenImage } from '@/components/TokenImage'
+import { getTokenByAddress } from '@/lib/tokens'
+import { useBuilderContext } from '@/app/collective-rewards/user/context/BuilderContext'
+import Link from 'next/link'
+
+const TableCellBase = ({
+  children,
+  className,
+  onClick,
+  columnId,
+  forceShow,
+}: HtmlHTMLAttributes<HTMLTableCellElement> & {
+  columnId: ColumnId
+  forceShow?: boolean
+}): ReactNode => {
+  const { columns } = useTableContext<ColumnId, TransactionHistoryCellDataMap>()
+  if (forceShow || !columns.find(col => col.id === columnId)?.hidden) {
+    return (
+      <td
+        className={cn('flex self-stretch items-center select-none', COLUMN_TRANSFORMS[columnId], className)}
+        onClick={onClick}
+      >
+        {children}
+      </td>
+    )
+  }
+  return null
+}
+
+const TableCell = ({
+  children,
+  className,
+  onClick,
+  columnId,
+  forceShow,
+}: HtmlHTMLAttributes<HTMLTableCellElement> & { columnId: ColumnId; forceShow?: boolean }): ReactNode => {
+  return (
+    <TableCellBase className={className} onClick={onClick} columnId={columnId} forceShow={forceShow}>
+      {children}
+    </TableCellBase>
+  )
+}
+
+interface CycleCellProps {
+  cycle: string | null
+}
+
+export const CycleCell: FC<CycleCellProps> = ({ cycle }): ReactElement => {
+  return (
+    <TableCell columnId="cycle" className="justify-center">
+      <Paragraph variant="body" className="text-v3-text-100 font-medium">
```

Comment:
You can alternatively add `bold` param and remove the redundant class names as in `<Paragraph variant="body" bold>`